packages/lib/ren-tx/src/configs/genericBurn.ts

```diff
@@ -113,128 +106,175 @@ const burnTransactionListener = (context: BurnMachineContext) => (
                             data: {
                                 ...tx,
+                                sourceTxConfs: confs,
+                                sourceTxConfTarget: tx.sourceTxConfTarget || 6,
                             },
                         });
-                    })
-                    .on("confirmation", burnListener)
-                    .catch((error) =>
-                        callback({ type: "BURN_ERROR", data: error }),
-                    );
-
-                const releaseListener = (status: string) => {
-                    status === "confirming"
-                        ? console.log(`confirming`)
-                        : console.log("status", status);
+                    }
                 };
 
-                const hashListener = (hash: string) => {
-                    callback({
-                        type: "CONFIRMED",
-                        data: { ...tx, renVMHash: hash },
-                    });
-                };
-
-                const transactionListener = (transaction: any) => {
-                    callback({
-                        type: "RELEASED",
-                        data: {
-                            ...tx,
-                            destTxHash: transaction.hash,
-                            // Can be used to construct blockchain explorer link
-                            rawDestTx: transaction,
-                        },
-                    });
-                };
-
-                const releaseRef = burn.release();
                 cleaners.push(() => {
-                    releaseRef._cancel();
-                    releaseRef.removeListener("status", releaseListener);
-                    releaseRef.removeListener(
-                        "transaction",
-                        transactionListener,
-                    );
-                    releaseRef.removeListener("txHash", hashListener);
-                });
-                releaseRef.catch((e) => {
-                    console.error("release error", e);
-                    callback({ type: "RELEASE_ERROR", data: new Error(e) });
+                    burnRef._cancel();
+                    burnRef.removeListener("confirmation", burnListener);
                 });
 
                 try {
-                    const res = await releaseRef
-                        .on("status", releaseListener)
-                        .on("transaction", transactionListener)
-                        .on("txHash", hashListener);
-                    callback({
-                        type: "RELEASED",
-                        data: { ...tx, renResponse: res },
-                    });
-                } catch (e) {
-                    callback({ type: "RELEASE_ERROR", data: new Error(e) });
+                    const r = await burnRef
+                        // host chain tx hash
+                        .on("transactionHash", (txHash: string) => {
```

Comment:
I'm working on standardizing the events emitted for mint-chain transactions (since right now its just forwarding web3 promievent events) - this probably won't be called `tranactionHash` to avoid confusion with RenVM txHashes. I'll let you know what it gets called.